src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -66,166 +61,222 @@ public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand,
+                .setMaxLength(19)
+                .setAllowedChars(TextFieldWidget2.NATURAL_NUMS)
+                .setPostFix(this.getPostFix()));
+        return sync;
+    }
+
     private String getMinValue() {
-        return String.valueOf(minEU);
+        return String.valueOf(minValue);
     }
 
     private String getMaxValue() {
-        return String.valueOf(maxEU);
+        return String.valueOf(maxValue);
     }
 
     private void setMinValue(String val){
-        try {
-            long c = Long.parseLong(val);
-            this.minEU = Math.min(maxEU - 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.minEU = Math.max(maxEU - 1, 0);
+        long parsedValue;
+
+        if (usePercent) {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MIN_PERCENT);
+        } else {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MIN_EU);
         }
+
+        this.minValue = Math.min(this.maxValue - 1, Math.max(0, parsedValue));
     }
 
     private void setMaxValue(String val){
-        try {
-            long c = Long.parseLong(val);
-            maxEU = Math.max(minEU + 1, c);
-        } catch (NumberFormatException e) {
-            GTLog.logger.warn(e);
-            this.maxEU = Math.max(minEU + 1, 2048);
+        long parsedValue;
+        long maxUpperLimit;
+
+        if (usePercent) {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MAX_PERCENT);
+            maxUpperLimit = 100;
+        } else {
+            parsedValue = GTUtility.tryParseLong(val, DEFAULT_MAX_EU);
+            maxUpperLimit = Long.MAX_VALUE;
         }
+
+        this.maxValue = Math.max(this.minValue + 1, Math.min(parsedValue, maxUpperLimit));
     }
 
     private boolean isInverted(){
-        return this.inverted;
+        return this.isInverted;
     }
 
     private void setInverted(boolean b){
-        this.inverted = b;
+        this.isInverted = b;
     }
 
-    @Override
-    public boolean canConnectRedstone() {
-        return true;
+    private boolean isUsePercent(){
+        return this.usePercent;
+    }
+
+    private void setUsePercent(boolean b){
+        this.usePercent =  b;
+        int length;
+
+        if (this.usePercent){ // using percent
+            this.minValue = DEFAULT_MIN_PERCENT;
+            this.maxValue = DEFAULT_MAX_PERCENT;
+            length = 3;
+        } else { // using discrete EU
+            this.minValue = DEFAULT_MIN_EU;
+            this.maxValue = DEFAULT_MAX_EU;
+            length = 19;
+        }
+
+        // update widgets
+        updateSyncedWidgets(length);
+    }
+
+    private void updateSyncedWidgets(int length) {
+        for (Widget widget : this.widgetsToUpdate.widgets) {
+            if (widget instanceof TextFieldWidget2) {
+                ((TextFieldWidget2) widget).setPostFix(null); // clear postfix
+                ((TextFieldWidget2) widget).setPostFix(this.getPostFix());
```

Comment:
First call is not needed anymore right?

Reply:
correct, the first instance of `postFix()` has been removed